schema.prisma

```diff
@@ -0,0 +1,682 @@
+model CourseCustomType {
+  /// 課程分類代碼， [A-Z]
+  code     String @id @db.VarChar(1)
+  /// 中文名稱
+  nameZhTw String @db.VarChar(50)
+  /// 英文名曾
+  nameEn   String @db.VarChar(50)
+}
+
+// ########################################
+//  領域專長
+// ########################################
+
+/// 領域專長說明主表
+model Domain {
+  /// 領域專長代碼
+  code   String  @id @db.VarChar(10)
+  /// 名稱
+  name   String  @db.VarChar(50)
+  /// 英文名稱
+  nameEn String  @db.VarChar(100)
+  /// 是否通過
+  apply  Boolean
+}
+
+/// 課程領域專長關聯
+model CourseDomain {
+  /// 課程識別碼
+  CourseId   String @db.VarChar(9)
+  /// 領域專長代碼
+  DomainCode String @db.VarChar(10)
+
+  @@unique([CourseId, DomainCode])
+}
+
+// ########################################
+// 課程大綱 from ceiba db
+// ########################################
+
+// 有許多重複的欄位，待確認此用方法
+// 有許多重複的欄位，待確認此用方法
+// 有許多重複的欄位，待確認此用方法
+
+/// 課程大綱主表
+model Syllabus {
+  // 原sn
+  id       Int    @id
+  /// 學期，如103-1
+  semester String @db.VarChar(6)
+  /// 課程識別碼
+  // CEIBA 課程網頁有非正式課程及正式課程兩種。非正式課程指不在正式課程中的課程。
+  // 正式課程，
+  // 課程識別碼如 101 80002
+
+  // 非正式課程，
+  // 課程識別碼，前三碼為XXX 80002
+  courseNo       String                 @db.VarChar(10)
+  /// 班級
+  classNo        String                 @db.VarChar(5)
+  /// CEIBA 專用之 6 碼課號
+  // 開設 CEIBA 網頁之課程皆有唯一的 6 碼課號。
+  // 未開設 CEIBA 課程網頁者，沒有。
+  courseSn       String?                @db.VarChar(6)
+  // lang 欄位為big5 時，格式為 第1週
+  // lang 欄位為 eng 時，格式為 Week 1
+  unit           String                 @db.VarChar(30)
+  /// 本筆課程進度表的日期
+  note           String                 @db.VarChar(256)
+  /// 是否為異動或新增
+  // 匯入之前學期的課程資料時填入 0 ，只要有異動或是新增時都填入 1 。 
+  // 區分課程進度的建立類型用。
+  flag           Boolean
+  /// 課程主題
+  subject        String
+  /// 課程進度表允許顯示的日期中的年月日。
+  // 過了設定的期限才會顯示給同學看見
+  publicDate     DateTime
+  /// 表格內容類型
+  classification SyllabusClassification
+  /// 檔案管理類型
+  fileManagement SyllabusFileManagement
+  lang           Language
+
+  // 1-1
+  officeTimeSwitch SyllabusOfficeTimeSwitch?
+
+  // 1-n
+  files    SyllabusFile[]
+  books    SyllabusBook[]
+  goals    SyllabusGoal[]
+  requests SyllabusRequest[]
+  officeTimes SyllabusOfficeTime []
+}
+
+/// 課程大綱附加資料
+model SyllabusFile {
+  /// 流水號，原sn
+  id         Int      @id
+  /// 課程大綱主表id，原syl_sn
+  syllabusId Int
+  /// 附加檔案名稱  
+  // 請用https://ceiba.ntu.edu.tw/course/
+  // 加上course_sn欄位的値
+  // 再加上/content/
+  fileName   String   @db.VarChar(255)
+  /// 語系
+  lang       Language
+
+  syllabus Syllabus @relation(fields: [syllabusId], references: [id])
+}
+
+model SyllabusBook {
+  /// 流水號，原sn
+  id         Int      @id
+  /// 大綱id
+  syllabusId Int
+  /// 書名
+  book       String
+  /// 語系
+  lang       Language
+
+  syllabus Syllabus @relation(fields: [syllabusId], references: [id])
+}
+
+/// 課程目標
+model SyllabusGoal {
+  /// 流水號
+  id             Int      @id
+  /// 大綱id
+  syllabusId     Int
+  /// 課程描述
+  describe       String
+  /// 課程要求
+  background     String
+  /// 課程目標
+  goal           String
+  /// 關鍵字
+  keyword        String
```

Comment:
Is this the keyword for searching and indexing?

Reply:
Not sure what it is.
@Klay24-Huang plz help XD

Reply:
This field is not use for searching.
I think it maybe diplay on course's syllabus ui, but UI/UX team is not design this ui yet.
I'm not sure this field will be used or not.